src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1104,10 +1238,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
         }
 
         protected virtual TagBuilder GenerateLink(
-            [NotNull] string linkText,
-            [NotNull] string url,
+            string linkText,
+            string url,
```

Comment:
I'll leave this as is for now and file a bug to change the design of anchor tag helper which includes a follow up to add a ArgNotNull here.